drivers/include/dht.h

```diff
@@ -56,11 +59,24 @@ typedef struct {
     uint16_t temperature;   /**< temperature in deci-Celsius */
 } dht_data_t;
 
+/**
+ * @brief   Device type of the DHT device
+ */
+typedef enum {
+    DHT11,                  /**< DHT11 device identifier */
+    DHT22,                  /**< DHT22 device identifier */
+    DHT21 = DHT22,          /**< DHT21 device identifier */
+    AM2301 = DHT22,         /**< AM2301 device identifier */
+} dht_type_t;
+
 /**
  * @brief   Configuration parameters for DHT devices
  */
 typedef struct {
     gpio_t pin;             /**< GPIO pin of the device's data pin */
+    dht_type_t type;        /**< type of the DHT device */
+    bool lsb_is_hundreds;   /**< 2022-12 aosong.com data sheet uses interprets
+                             *   low bits as 0.01°C per LSB */
```

Comment:
Are you sure? The example in the data sheet is

```
-10.1°C = 0b00001010, 0b10000001
```

That would mean the low byte is in 0.1 °C

Reply:
OK, I see. I wonder why they couldn't have called that device DHT12 :/

Reply:
i think there is a dht12 (featuring i2c)

Reply:
Or DHT11B or DHT11+ or DHT11v2 or DHT11 rev 2 or whatever. IMO a breaking change in the wire interface shouldn't need a look at the production timestamp to detect.

Reply:
See above: https://github.com/RIOT-OS/RIOT/pull/19718#discussion_r1226569173